src/kv/test/kv_test.cpp

```diff
@@ -745,4 +760,74 @@ TEST_CASE("private recovery map swap")
       // impossible during recovery.
     }
   }
+}
+
+TEST_CASE("replicated and derived table serialisation")
+{
+  auto encryptor = std::make_shared<ccf::NullTxEncryptor>();
+  Store store;
+  store.set_encryptor(encryptor);
+
+  auto& data_replicated = store.create<size_t, size_t>(
+    "data_replicated", kv::SecurityDomain::PUBLIC, true);
+  auto& data_derived = store.create<size_t, size_t>(
+    "data_derived", kv::SecurityDomain::PUBLIC, false);
+  auto& data_replicated_private = store.create<size_t, size_t>(
+    "data_replicated_private", kv::SecurityDomain::PRIVATE, true);
+  auto& data_derived_private = store.create<size_t, size_t>(
+    "data_derived_private", kv::SecurityDomain::PRIVATE, false);
+
+  {
+    Store::Tx tx(store.next_version());
+    auto data_view = tx.get_view(data_replicated);
+    auto data_view_private = tx.get_view(data_replicated_private);
+    data_view->put(42, 42);
+    data_view_private->put(43, 43);
+    auto [success, reqid, serialised] = tx.commit_reserved();
+    REQUIRE(success == kv::CommitSuccess::OK);
+
+    REQUIRE(
+      store.deserialise(serialised.replicated) == kv::DeserialiseSuccess::PASS);
+
+    REQUIRE(serialised.replicated.size() > 0);
+    REQUIRE(serialised.derived.size() == 0);
+  }
+
+  {
+    Store::Tx tx(store.next_version());
+    auto data_view = tx.get_view(data_derived);
+    auto data_view_private = tx.get_view(data_derived_private);
+    data_view->put(42, 42);
+    data_view_private->put(43, 43);
+    auto [success, reqid, serialised] = tx.commit_reserved();
```

Comment:
Should we try to use `commit_reserved()` as little as possible here? You could get the serialised data by attaching the `StubConsensus` to the store.

Reply:
I do not get why. `commit_reserved` will call `tx::serialise()` which is what I want to test here, there is no consensus involved.